llms/mlx_lm/tuner/dora.py

```diff
@@ -75,16 +83,57 @@ def __init__(
             shape=(input_dims, r),
         )
         self.lora_b = mx.zeros(shape=(r, output_dims))
-        self.m = mx.linalg.norm(self.linear.weight, axis=1)
+
+    def set_linear(self, linear):
+        """
+        Set the self.linear layer and recompute self.m with respect to quantization
+        """
+        self.linear = linear
+        # self.m needs to be recomputed when self.linear changes
+        # which doesn't happen in from_linear when dora_lin.linear = linear
+        self.m = mx.linalg.norm(self._dequantized_weight(), axis=1).astype(mx.float32)
```

Comment:
I don't think you need the `astype` cast here.

Reply:
On second thought maybe it's a good idea for this parameter to have higher precision (but then we need to be careful with the types in the `__call__` function as per my comment.

Reply:
I can't quite recall exactly what the issue was. I wrote a test with back prop and I was getting `NaN` gradients. I narrowed it down to an issue with a type not being high enough precision but I'm not sure if it was the `self.m` parameter that was causing the issue. I just removed the casting to that type and there is no issue. 🤷